### DISCUSSION TITLE
router-link not resolved in unit test

### DISCUSSION THREAD

**Discussion by julisch94** [vuejs/router#2184] [Help and Questions] [answered]
Hey folks, I would appreciate some help with this.
I'm in the process of upgrading `vue@3.4.15` to `vue@3.4.16` but with this some of my tests that include `router-link` break.

I'm using `@testing-library/vue@8.0.3` to render my component like this:
```ts
    const router = createRouter({
      history: createWebHashHistory(),
      routes,
    })

    const screen = render(OverviewPage, {
      global: {
        plugins: [router],
      },
    })

    screen.debug()
```

Before, it printed like this:
```html
<body>
  <a
    class=""
    href="#/?locale=de-CH"
    hreflang="de"
  >
    German
  </a>
  <a
    class=""
    href="#/?locale=fr-CH"
    hreflang="fr"
  >
    French
  </a>
  <a
    class=""
    href="#/?locale=it-CH"
    hreflang="it"
  >
    Italian
  </a>
</body>
```

Now, it prints this:
```html
<body>
  <router-link
    hreflang="de"
    to="/?locale=de-CH"
  >
    German
  </router-link>
  <router-link
    hreflang="fr"
    to="/?locale=fr-CH"
  >
    French
  </router-link>
  <router-link
    hreflang="it"
    to="/?locale=it-CH"
  >
    Italian
  </router-link>
</body>
```

I really don't know if I'm doing anything wrong. Why is the `router-link` no longer resolved as before?
I noticed I'm getting a warning for every `router-link` on my page when running my test case:
```
[Vue warn]: resolveComponent can only be used in render() or setup().
[Vue warn]: resolveComponent can only be used in render() or setup().
[Vue warn]: resolveComponent can only be used in render() or setup().
```

Dependencies:
```
"vue-router": "4.2.5"
"@testing-library/vue": "8.0.3",
"@testing-library/dom": "9.3.4",
"@vue/test-utils": "2.4.5",
"vite": "5.0.12",
"vitest": "1.2.1",

**Comment by skirtles-code** [comment, marked as answer]
The warning you mentioned would seem to indicate that you have two copies of Vue included in your project. One copy is providing the `resolveComponent` function, while the other is running the component rendering. That leads to `resolveComponent` believing it is being run outside of `render`.

You could try deleting `node_modules` and reinstalling, though that may not help if both versions are listed in the lock file. Commands like `npm list vue` and `npm why vue` could also help to establish what copies of Vue are present.

**Reply by julisch94** [reply]
Thank you! Your hint was very helpful. I was able to confirm that different versions of Vue were used at the same time with `npm list vue`.

Adding the following lines in the root `package.json` of my monorepo solved my issue:
```json
"peerDependencies": {
    "vue": "3.4.16"
  }
```
